### PR TITLE
common : factorize IPv4/v6 normalize IP/Address

### PR DESCRIPTION
**Pull request by nplanel**
Signed-off-by: Nicolas PLANEL <nplanel@redhat.com>

### CODE REVIEW

File: common/types.go [hunk 1]
```diff
@@ -225,8 +225,20 @@ func IPv6Supported() bool {
 }
 
 // NormalizeIPForUrl returns a string normalized that can be used in URL. Brackets
-// will be used for IPV6 addresses.
-func NormalizeIPForUrl(ip net.IP) string {
+// will be used for IPV6 addresses. Accept net.IP of string as input.
```
Comment (masco-enovance) [inline common/types.go]: nit: s/of/or

File: analyzer/flow_client.go [hunk 2]
```diff
@@ -110,7 +110,7 @@ func (c *FlowClientWebSocketConn) Close() error {
 // Connect to the WebSocket flow server
 func (c *FlowClientWebSocketConn) Connect() error {
 	authOptions := NewAnalyzerAuthenticationOpts()
-	authAddr := common.NormalizeAddrForURL(c.url.Hostname())
+	authAddr := common.NormalizeIPForUrl(c.url.Hostname())
```
Comment (safchain) [inline analyzer/flow_client.go]: Hostname can return a domain name and even if we resolve the address earlier in the call flow I would prefer to keep addr to support domain name.

File: common/types.go (hunk 1 above)
Comment (nplanel) [reply common/types.go]: done

File: analyzer/flow_client.go (hunk 2 above)
Comment (nplanel) [reply analyzer/flow_client.go]: done

**Comment by nplanel** [comment]
run skydive-scale-tests

**Pull request closed without merging**